slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -313,34 +346,18 @@ void SLAMServiceImpl::CreateMap() {
             set_start_time = true;
         }
 
-        int num_nodes;
         {
             std::lock_guard<std::mutex> lk(map_builder_mutex);
             auto measurement = map_builder.GetDataFromFile(file);
             if (measurement.ranges.size() > 0) {
                 trajectory_builder->AddSensorData(kRangeSensorId.id,
                                                   measurement);
-                num_nodes = map_builder.map_builder_->pose_graph()
-                                ->GetTrajectoryNodes()
-                                .size();
-
-                auto local_poses = map_builder.GetLocalSlamResultPoses();
-                if (local_poses.size() > 0) {
-                    auto latest_local_pose = local_poses.back();
-                    cartographer::transform::Rigid3d global_pose =
-                        map_builder.GetGlobalPose(trajectory_id,
-                                                  latest_local_pose);
-                    myfile << "global_pose: " << global_pose.DebugString()
-                           << std::endl;
-                }
```

Comment:
Let's set `latest_global_pose` as a member variable within the slam service that's continuously updated here every time we're updating the map. We can then use that global pose when we want to read & return it in `GetPosition`

Reply:
Kk so make pose global and trajectory_id local

Reply:
I agree here. This is what we do for orbslam as well. We keep the map and position as member variables and update them as part of mapping. That way calls to GetMap and GetPosition can use the previous map/position if mapping is in progress.